Source/Core/UICommon/NetPlayIndex.cpp

```diff
@@ -150,9 +153,8 @@ void NetPlayIndex::NotificationLoop()
       m_running.Set(false);
       return;
     }
-
-    Common::SleepCurrentThread(1000 * 5);
-  }
+  } while (m_running.IsSet() &&
+           m_session_thread_cv.wait_for(lock, std::chrono::seconds(5)) == std::cv_status::timeout);
```

Comment:
It looks like a spurious wakeup would cause this thread to end prematurely.

Reply:
What would cause a spurious wakeup?